redash/query_runner/elasticsearch.py

```diff
@@ -197,34 +197,53 @@ def collect_value(mappings, row, key, value, type):
             add_column_if_needed(mappings, key, key, result_columns, result_columns_index)
             row[key] = value
 
-        def collect_aggregations(mappings, rows, parent_key, data, row, result_columns, result_columns_index):
-            if isinstance(data, dict):
-                for key, value in data.iteritems():
-                    val = collect_aggregations(mappings, rows, parent_key if key == 'buckets' else key, value, row, result_columns, result_columns_index)
-                    if val:
-                        row = get_row(rows, row)
-                        collect_value(mappings, row, key, val, 'long')
-
-                for data_key in ['value', 'doc_count']:
-                    if data_key not in data:
-                        continue
-                    if 'key' in data and len(data.keys()) == 2:
-                        key_is_string = 'key_as_string' in data
-                        collect_value(mappings, row, data['key'] if not key_is_string else data['key_as_string'], data[data_key], 'long' if not key_is_string else 'string')
+        def parse_bucket_to_row(mappings, data, row, agg_key):
+            sub_agg_key = ""
+            for key, item in data.iteritems():
+                if key == 'key_as_string':
+                    continue
+                if key == 'key':
+                    if 'key_as_string' in data:
+                        collect_value(mappings, row, agg_key, data['key_as_string'], 'string')
+                    else:
+                        collect_value(mappings, row, agg_key, data['key'], 'string')
+                    continue
+
+                if type(item) is str:
```

Comment:
using `isinstance(item, str)` may be better; as also in all below type checks

Reply:
Done